perceiver/model/image/optical_flow.py

```diff
@@ -0,0 +1,227 @@
+import os
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+from transformers import PerceiverConfig as HuggingfacePerceiverConfig, PerceiverForOpticalFlow
+from transformers.models.perceiver.modeling_perceiver import PerceiverImagePreprocessor, space_to_depth
+
+from perceiver.model.core import (
+    EncoderConfig,
+    InputAdapter,
+    OutputAdapter,
+    PerceiverConfig,
+    PerceiverDecoder,
+    PerceiverEncoder,
+    PerceiverIO,
+)
+from perceiver.model.core.config import DecoderConfig
+from perceiver.model.core.convert import (
+    copy_cross_attention_layer_params,
+    copy_param,
+    copy_params,
+    copy_self_attention_block_params,
+)
+from perceiver.model.image.common import FourierPositionEncoding
+
+
+@dataclass
+class OpticalFlowEncoderConfig(EncoderConfig):
+    image_shape: Tuple[int, int] = (368, 496)
+    num_temporal_channels: int = 2
+    num_image_input_channels: int = 27
+    num_image_output_channels: int = 64
+    num_frequency_bands: int = 64
+    spatial_downsample: int = 1
+    temporal_downsample: int = 2
```

Comment:
I would prefer the usage of `num_patch_input_channels` as the approach using `patch_size` would only work for a patch-size of `3` since the 3 image channels in the input are fixed (e.g. for a patch-size of `5` a configuration on `75` channels would be required as opposed to `5**3=125`). Or did I misunderstand the proposed concept of `patch_size` here?

Reply:
Good catch and sorry for the confusion. This makes sense of course!